sklearn/metrics/_dist_metrics.pyx

```diff
@@ -295,11 +307,49 @@ cdef class DistanceMetric:
         This can optionally be overridden in a base class.
 
         The rank-preserving surrogate distance is any measure that yields the same
-        rank as the distance, but is more efficient to compute. For example, for the
-        Euclidean metric, the surrogate distance is the squared-euclidean distance.
+        rank as the distance, but is more efficient to compute. For example, the
+        rank-preserving surrogate distance of the Euclidean metric is the
+        squared-euclidean distance.
         """
         return self.dist(x1, x2, size)
 
+    cdef DTYPE_t csr_dist(
```

Comment:
I do not think we need `csr_dist` and `csr_rdist` yet since we do not have sparse support in this PR?